pkg/reconciler/v1alpha1/namespace/namespace.go

```diff
@@ -18,26 +18,114 @@ package namespace
 
 import (
 	"context"
-
 	"github.com/knative/eventing/contrib/gcppubsub/pkg/util/logging"
 	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
 	"go.uber.org/zap"
+	"k8s.io/api/core/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 const (
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "knative-eventing-namespace-controller"
+
 	defaultBroker             = "default"
 	knativeEventingAnnotation = "eventing.knative.dev/injection"
 
-	// Name of the corev1.Events emitted from the reconciliation process
+	// Name of the corev1.Events emitted from the reconciliation process.
 	brokerCreated = "BrokerCreated"
```

Comment:
Maybe to keep consistency with the other controllers, we can have BrokerReconciled and emit the event in the caller not callee; in the reconcile function. Just a suggestions, you can skip it if you want.

Reply:
Just check that the yaml verbs are properly set to emit events

Reply:
I can't put my finger on why, but I think I prefer it this way. My hunch is that we will add events for failing to create or get the Broker too, so the extra specificity is something I like.

That's not to say I can't be convinced, but right now I think I prefer this way. If others feel the way you do, I'll change it.